globus_sdk/authorizers/refresh_token.py

```diff
@@ -1,19 +1,12 @@
 import warnings
 import time
+
 from globus_sdk.authorizers.base import GlobusAuthorizer
 
-# a percent error expected in the `expires_in` values on AccessTokens,
-# expressed as a fractional value
-# taken out of the expiration value, not the total time (because that would be
-# weird, yeah?)
-# This is 5%, not 0.05%!
-_EXPIRATION_PERCENT_ERROR = 0.05
-# this is a fixed number of seconds to be pulled down from the expiration
-# timer, so that when the expires_in value is very small, the percent error
-# doesn't go to 0 and make us run the timer all the way down
-# so, we want (NOW - expires_in - (expires_in*PCT_ERROR) - FIXED_ERROR)
-# ultimately
-_EXPIRATION_FIXED_ERROR = 5
+
+# Provides a buffer for token expiration time to account for
+# possible delays or clock skew.
+EXPIRES_ADJUST_SECONDS = 60
```

Comment:
We're not worried about (stable) clock skew. That's solved by returning `expires_in` rather than `expires_at`; Auth can think it's 1970 for all you care. What we are worried about is:

- Clock drift, but this is probably mostly a theoretical worry with modern hardware over the 48 hour lifetime of a token.
- Network delays. Both:
  - Time from token minting in Auth until the response is processed in the client.
  - Time from "is expiry time good?" check in the client until "is expiry time good?" check in Auth.

The two network delays can potentially add up to more than 60 seconds. The only way to get a truly robust value for  `EXPIRES_ADJUST_SECONDS` is to force a specific `timeout` for both these calls and setting the adjustment parameter to `timeout*2 + fudge_factor_for_potential_clock_drift_or_whatever_just_in_case`.

Now, 60 seconds is clearly sufficient in the vast majority of cases, but someone will run into this, potentially many someones if we have problems on the Auth side.

(We can still run into someone actually changing the clock of a node by some significant factor, but there's no programmatic way of guarding against that so we don't worry about it.)